server/src/main/java/org/elasticsearch/readiness/ReadinessService.java

```diff
@@ -277,7 +279,20 @@ private boolean isMasterElected(ClusterState clusterState) {
     // protected to allow mock service to override
     protected boolean areFileSettingsApplied(ClusterState clusterState) {
         ReservedStateMetadata fileSettingsMetadata = clusterState.metadata().reservedStateMetadata().get(FileSettingsService.NAMESPACE);
-        return fileSettingsMetadata != null && fileSettingsMetadata.version().equals(ReservedStateMetadata.NO_VERSION) == false;
+        if (fileSettingsMetadata == null) {
+            // In order to block readiness on file settings being applied, we need to know that the master node has written an initial
```

Comment:
Maybe worth a comment re-iterating that this is only if the metadata is not found, indicating it is <8.4 anyway